src/core/lv_event.c

```diff
@@ -419,20 +431,41 @@ static lv_res_t event_send_core(lv_event_t * e)
     lv_indev_t * indev_act = lv_indev_get_act();
     if(indev_act) {
         if(indev_act->driver->feedback_cb) indev_act->driver->feedback_cb(indev_act->driver, e->code);
+        if(e->stop_processing) return LV_RES_OK;
         if(e->deleted) return LV_RES_INV;
     }
 
     lv_res_t res = LV_RES_OK;
-    res = lv_obj_event_base(NULL, e);
-
     lv_event_dsc_t * event_dsc = res == LV_RES_INV ? NULL : lv_obj_get_event_dsc(e->current_target, 0);
 
     uint32_t i = 0;
     while(event_dsc && res == LV_RES_OK) {
-        if(event_dsc->cb && (event_dsc->filter == LV_EVENT_ALL || event_dsc->filter == e->code)) {
+        if(event_dsc->cb  && ((event_dsc->filter & LV_EVENT_PREPROCESS) == LV_EVENT_PREPROCESS)
+           && (event_dsc->filter == LV_EVENT_ALL || (event_dsc->filter & ~LV_EVENT_PREPROCESS) == e->code)) {
             e->user_data = event_dsc->user_data;
             event_dsc->cb(e);
 
+            if(e->stop_processing) return LV_RES_OK;
```

Comment:
I'm not sure I like the idea of one event handler being able to cancel another one. IMO `stop_processing` should be named `prevent_default_processing` (or `prevent_def_proc`) and should only prevent `lv_obj_event_base` from being called.  Other custom event handlers can read the value of `e->prevent_def_proc` and skip running their logic if it's considered "default".

The current behavior allows one event handler to completely disable another one which seems dangerous and hard to predict.

Reply:
I agree with @X-Ryl669. Let's say I add preprocessed click event to a button and on the first click I create a pop up saying "Are you sure?" and prevent other click events to run.

Reply:
Fair enough; let's do it this way.